src/coreclr/src/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/NonGCStaticsNode.cs

```diff
@@ -136,13 +136,13 @@ public override ObjectData GetData(NodeFactory factory, bool relocsOnly)
             if (factory.PreinitializationManager.HasLazyStaticConstructor(_type))
             {
                 int alignmentRequired = Math.Max(_type.NonGCStaticFieldAlignment.AsInt, GetClassConstructorContextAlignment(_type.Context.Target));
-                int classConstructorContextStorageSize = GetClassConstructorContextStorageSize(factory.Target, _type);
+                int classConstructorContextStorageSize = GetClassConstructorContextSize(factory.Target, _type);
                 builder.RequireInitialAlignment(alignmentRequired);
                 
-                Debug.Assert(classConstructorContextStorageSize >= GetClassConstructorContextSize(_type.Context.Target));
+                Debug.Assert(classConstructorContextStorageSize >= GetClassConstructorContextStorageSize(_type.Context.Target, _type));
```

Comment:
This too.

Reply:
Could you undo this change? (The assert doesn't make sense like this because it's comparing two of the same values.)